[READY] Cool & good new arcade game

PKPenguin321 opened the pull request:
:cl:
add: Keep your eye out for a rare new arcade game!
/:cl:

https://tgstation13.org/phpBB/viewtopic.php?f=9&t=32&view=unread#p484730

Has a 2/100 chance of spawning instead of a regular arcade game

PKPenguin321 committed:
cool & good new arcade game

PKPenguin321 commented on code/game/machinery/computer/arcade.dm:
```diff
@@ -1112,6 +1124,34 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
+/obj/machinery/computer/arcade/amputation/attack_hand(mob/user)
+	if(!iscarbon(user))
+		return
+	var/mob/living/carbon/c_user
```
= user

Shadowflame909 commented:
Both of the arcade machines just make bombs as a prize once emagged.


Why not have this replace one of them?

Preferably the Outbomb Cuban Pete variant, because that ones less creative

Farquaar commented:
Any screenshots of the sprite? Or will it look the same?

SpadesNeil commented:
![image](https://user-images.githubusercontent.com/6484985/54746519-b978d600-4ba2-11e9-8961-52d4235af34a.png)

From the forum.

wesoda25 commented:
Then we can add the one armed bandit.

anconfuzedrock commented:
part of me wants this to be an emag function since outbomb is abused so easily

Dorsisdwarf commented:
How does this interact with people who have more than 2 hands?

SpadesNeil commented:
>How does this interact with people who have more than 2 hands?

You get more than two tries.

I think.

Idk, I didn't code it.

PKPenguin321 commented:
>How does this interact with people who have more than 2 hands?

Probably ignores all the extra hands

Qustinnus commented on code/game/machinery/computer/arcade.dm:
```diff
@@ -65,7 +65,19 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
+		var/list/normalgames = list()
+		var/list/raregames = list()
+		for(var/games in subtypesof(/obj/item/circuitboard/computer/arcade))
+			var/obj/item/circuitboard/computer/arcade/thegame = games//you just lost
```
die

PKPenguin321 replied:
Owned

Qustinnus commented on code/game/machinery/computer/arcade.dm:
```diff
@@ -65,7 +65,19 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
+		if(prob(1))
+			choice = pick(raregames)
+		else
+			choice = pick(normalgames)
```
just make a weighted light and use pickweight, this is gay

PKPenguin321 replied:
Oh, good idea

PKPenguin321 commented on code/game/machinery/computer/arcade.dm:
```diff
@@ -1112,6 +1124,34 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
+	if(do_after(c_user, 50, target = src))
+		to_chat(c_user, "<span class='userdanger'>The guillotine drops on your arm, and the machine sucks it in!</span>")
+		if(l_arm)
+			l_arm.dismember()
```
was too lazy to dig into dismemberment procs and make that happen. is there a good way to get the actual limb from the active hand?

tralezab replied:
Active hand index is always odd for left hands, and always even for right. You can grab the limb depending on whether it is even or odd. if you wanted to get funky the indexes come in sets of 2 but i wouldn't even bother, nobody has used extra hands in the code (outside of admin fun) for the ~3 years it has been in

PKPenguin321 replied:
nice, got it working

PKPenguin321 committed:
fixes issues, code improvements
- arcade machine board selection now uses pickweight
- the amputation machine actually works and has sound effects now
- removes raregame var and base path for arcade boards

deathride58 commented on code/game/machinery/computer/arcade.dm:
```diff
@@ -1112,6 +1115,36 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
+			l_arm.dismember()
+			qdel(l_arm)
+		else
+			r_arm.dismember()
```
This will cause a runtime if the user lacks a right arm

PKPenguin321 replied:
Good point. Would prefer to fix it by making this whole thing just target the active arm, but again, there doesn't seem to be a way to do that...

PKPenguin321 committed:
more comprehensive arm checks

PKPenguin321 committed:
better arm detection

amputation game will now target the active hand first
arm checks improved to not be redundant
saved a var definition in the prize vending code

PKPenguin321 commented:
ready to go

SpaceManiac reviewed (approved):
Tested, interacts sanely with >2 hands
